### PR TITLE
docs: Docs automation

### PR DESCRIPTION
**Pull request by Eric-Warehime**
## Summary

Adds a hidden command to conduit which will generate the cobra CLI docs.

Also adds a new `conduit-docs` command which will be used via `go:generate` on config files to generate the markdown pages for our developer portal.  
This portion is still WIP, I'll include renders of some of the more interesting configs people can see what they look like for a given revision.

For now I've just got the config graphs working. For example, the filter processor page is as follows:

Sample docs page generated with `conduit-docs`. Also see https://github.com/algorand/docs/pull/964 for the docs pages.
Screenshot 2022-12-20 at 9 26 27 AM

### CODE REVIEW

**Comment by Eric-Warehime** [comment]
Some caveats...There's a tradeoff that I've made by parsing the AST instead of using runtime reflection. It enables us to grab comments from the code (which I believe are erased at runtime), but it makes it pretty difficult to parse objects in other packages like `expression.FilterType`. 

It also means we can't use `go:embed` to load the sample config--although I think putting examples in comments as I've done for filter processor is a decent substitute for this.

File: Makefile [hunk 1]
```diff
@@ -25,9 +25,12 @@ export GO_IMAGE = golang:$(shell go version | cut -d ' ' -f 3 | tail -c +3 )
 # This is the default target, build everything:
 all: conduit cmd/algorand-indexer/algorand-indexer go-algorand idb/postgres/internal/schema/setup_postgres_sql.go idb/mocks/IndexerDb.go
 
-conduit: go-algorand
+conduit: go-algorand conduit-docs
 	go generate ./... && cd cmd/conduit && go build -ldflags="${GOLDFLAGS}"
 
+conduit-docs:
+	go install ./cmd/conduit-docs/
```
Comment (winder) [inline Makefile:32]: I think you can use `go run ./cmd/conduit-docs` in the go-generate command to avoid needing an install step.

Comment (shiqizng) [reply Makefile:32]: are these docs for dev portal or are they readmes staying in the conduit repo?

**Comment by shiqizng** [comment]
this is great! is it be possible to fix the table width so the type is in a single line.  Right now, the type value for filters look confusing.

File: Makefile (hunk 1 above)
Comment (Eric-Warehime) [reply Makefile:32]: I intend them for the dev portal, but we could use them for our readmes as well--there is obviously a lot of overlap.

Is it bad to use `go install`? I've wondered why we don't have an install target for indexer in our makefile.

**Review by AlgoStephenAkiki — APPROVED** [review]

**Pull request merged**